internal/yggdrasil/yggdrasil.go

```diff
@@ -55,6 +62,24 @@ func NewYggdrasilHandler(deviceRepository edgedevice.Repository, deploymentRepos
 	}
 }
 
+func isRegisteredURL(url *url.URL) bool {
+	parts := strings.Split(url.Path, "/")
+	if len(parts) == 0 {
+		return false
+	}
+
+	last := parts[len(parts)-1]
+	return last == "registration"
+}
+
+func (h *Handler) SetAuthType(r *http.Request) int {
```

Comment:
WDYT about `GetAuthType` instead? And maybe moving it to a separate file? This method is not part of the HTTP API after all.

Reply:
To be checked on ECOPROJECT-401

Reply:
What do you mean by that?